### ISSUE TITLE
create worker, work and die )

### ISSUE THREAD

**Issue by blackcatt** [walkor/workerman#875]
Hi, i need to create worker **inside other worker** and then correctly stop(destroy) it after it do some job.
something like that:
```
$small_worker = new Worker();
$small_worker->onWorkerStart = function() use (&$small_worker) {
  echo "I've worked and that's enough.\n";
  //$small_worker->stop(); //does not work (
  //Worker::stop(); //does not work (
  die(); // work, but i'm not sure if it's right
}
```
Thanks

**Comment by walkor** [comment]
Using `die()` will generate a log. Use `Worker::stopAll()` instead of `die()`.

**Comment by blackcatt** [comment]
Walkor, I used this code ( Worker::stopAll(); ) and it seemed to me that everything works fine. But it's not.
stopAll() will stop mу small_worker AND parent worker too.
But as i wrote i need start small_worker, do some job asynchronously from parent in onWorkerStart and end it **without** stoppping parent worker.

**Comment by indigoram89** [comment]
@walkor  good question but no answer is here =)

**Comment by walkor** [comment]
Workerman does not support dynamic process creation.

**Comment by indigoram89** [comment]
@walkor but we have solution to stop the current child worker only?

**Comment by xpader** [comment]
As I know, stopAll() is not stop current worker, is not just stop current process, it is stop the current process and restart a new process. You can think of it as restarting the current process

**Comment by indigoram89** [comment]
Yes, but I need only stop and restart current worker process, for example I have two workers and I would like to stop the first of them, something like `if ($worker->id === 1) $worker->stop();` then master process will restart it automatically.
